bolt-modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -117,10 +118,26 @@ def run_task_raw(task_name, targets, description = nil, task_args = nil, &block)
       end || (raise with_stack(:TYPE_MISMATCH, 'Task parameters do not match'))
 
       task = task_signature.task
+
+      # Wrap parameters marked with '"sensitive": true' in the task metadata with a
+      # Sensitive wrapper type. This way it's not shown in logs
+      use_args.each do |k, v|
+        if task.parameters and task.parameters[k]['sensitive']
+          use_args[k] = Puppet::Pops::Types::PSensitiveType::Sensitive.new(v)
+        end
+      end
+
     end
 
-    unless Puppet::Pops::Types::TypeFactory.data.instance?(use_args)
-      raise with_stack(:TYPE_NOT_DATA, 'Task parameters is not of type Data')
+    unless Puppet::Pops::Types::TypeFactory.rich_data.instance?(use_args)
```

Comment:
If we put this check inside the 124 loop and check each param instead of the entire params hash we can support sensitive from plans without opening up tasks to the entire world of richdata which will probably require more work.

Reply:
Ah yeah. we should move this check to that branch and have a per param check in the "local code" branch to account for sensitive.